pkg/update_pod_status_util.go

```diff
@@ -212,22 +212,36 @@ func runBlkdiscard(logger logr.Logger, cmd []string, wg *sync.WaitGroup, guard c
 	<-guard
 }
 
-func isVolNeedInitialisation(initializedVolumes []string, volName, pvcUID string) bool {
+func isVolInitialisationNeeded(logger logr.Logger, initializedVolumes []string, volName,
+	pvcUID string) (needInit bool, volumes []string) {
 	for idx := range initializedVolumes {
 		initVolInfo := strings.Split(initializedVolumes[idx], "@")
 		if initVolInfo[0] == volName {
-			if len(initVolInfo) < 2 || initVolInfo[1] == pvcUID {
-				return false
+			if len(initVolInfo) < 2 {
+				initializedVolumes = append(initializedVolumes, fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID))
+
+				return false, remove(initializedVolumes, initializedVolumes[idx])
```

Comment:
We can directly overwrite the index as we not initialising here
initializedVolumes[idx] = fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID)